packages/jtex/src/download.ts

```diff
@@ -47,13 +50,21 @@ function defaultUrl(session: ISession, template: string) {
   return `${session.API_URL}/templates/${template}`;
 }
 
-function defaultPath(template: string, hash: boolean, buildDir?: string) {
+function defaultPath(
+  template: string,
+  hash: boolean,
+  opts: { kind?: TemplateKinds; buildDir?: string },
+) {
+  const { kind, buildDir } = opts;
   const subdirs: string[] = [];
   if (buildDir) subdirs.push(buildDir);
-  subdirs.push(
-    'templates',
-    hash ? createHash('sha256').update(template).digest('hex') : join(...template.split('/')),
-  );
+  subdirs.push('templates');
+  if (hash) {
+    if (kind) subdirs.push(kind);
+    subdirs.push(createHash('sha256').update(template).digest('hex'));
```

Comment:
What is this hash for? Is this where the template is downloaded to?

Reply:
Yes, if the `template` specified is a URL, it is downloaded to a URL hash folder (rather than trying to extract a nice-looking folder path from an arbitrary URL).

We could have like a special case for github urls or something, if that gets matched, extract `org/repo`? But there needs to be some fallback that works for all URLs.

Reply:
Ah ok, clever, yeah that is totally fine. People won't be looking in that folder I don't think, so this should be fine!